src/state/transactions/hooks.tsx

```diff
@@ -113,9 +113,24 @@ export function useHasPendingApproval(token?: Token, spender?: string): boolean
           return false
         } else {
           if (tx.info.type !== TransactionType.APPROVAL) return false
-          return tx.info.spender === spender && tx.info.tokenAddress === token.address && isTransactionRecent(tx)
+          const isPendingApproveOrRevoke =
+            tx.info.spender === spender && tx.info.tokenAddress === token.address && isTransactionRecent(tx)
+          if (approvalType === 'revoke') {
+            return BigNumber.from(tx.info.amount)?.eq(0) && isPendingApproveOrRevoke
+          } else {
+            return BigNumber.from(tx.info.amount)?.gt(0) && isPendingApproveOrRevoke
+          }
         }
       }),
-    [allTransactions, spender, token?.address]
+    [allTransactions, approvalType, spender, token?.address]
   )
 }
+
+// returns whether a token has a pending approval transaction
+export function useHasPendingApproval(token?: Token, spender?: string): boolean {
+  return useHasPendingApprovalChange(token, spender, 'approve')
```

Comment:
The above does not make use of `'revoke' | 'approve'`, but treats it as a boolean. Could you simplify the logic by refactoring `useHasPendingApprovalChange` to `usePendingApproval`, which returns the `tx.info.amount` being approved, and then check if it is `eq` or `gt` 0 within these derived functions? I think eliminating the branching from the helper function will make this whole file easier to read.

Reply:
done